source/Coop.Core/Server/Services/Sync/Handlers/PeerQueueOverloadedHandler.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Network;
+using Coop.Core.Server.Services.Time.Messages;
+using GameInterface.Services.GameDebug.Messages;
+using GameInterface.Services.Heroes.Enum;
+using GameInterface.Services.Heroes.Messages;
+using LiteNetLib;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Coop.Core.Server.Services.Sync.Handlers
+{
+    internal class PeerQueueOverloadedHandler : IHandler
+    {
+        private readonly IMessageBroker messageBroker;
+        private readonly INetwork network;
+
+        private readonly Poller poller;
+        private readonly HashSet<NetPeer> overloadedPeers;
+
+        private TimeControlEnum originalSpeed;
+
+        private readonly TimeHandler timeHandler;
+
+        public PeerQueueOverloadedHandler(
+            IMessageBroker messageBroker, 
+            INetwork network,
+            TimeHandler timeHandler
+        )
+        {
+            this.messageBroker = messageBroker;
+            this.network = network;
+            this.timeHandler = timeHandler;
+
+            overloadedPeers = new();
+            poller = new(Poll, TimeSpan.FromMilliseconds(100));
+
+            messageBroker.Subscribe<PeerQueueOverloaded>(Handle);
+
+            poller.Start();
```

Comment:
start polling when overloadedPeers is > 0 and stop on <= 0

Reply:
We do not need to start polling here since its now started [here](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/pull/713/files#diff-b7d2c2ceec17a26ba8e54aa12fafe21a8c0e1e0d6f561b2adbb9e60b1412a903R65)